Marlin/Marlin_main.cpp

```diff
@@ -499,27 +497,28 @@ static bool drain_queued_commands_P() {
  */
 void enqueuecommands_P(const char* pgcode) {
   queued_commands_P = pgcode;
```

Comment:
To make this really work, in every case, we need a queue here. 
The current, and former, behavior was, and is, that a new series of commands is replacing the old one, even if unfinished. Throwing an error, or at least a warning, if queued_commands_P != NULL would also be correct if we want to finish the old sequence first. But the warning 4 lines above should be enough.
The difference between then and now is only that the first command of the sequence is not queued into the B-buffer immediately. It will be drained at the beginning of the next get_comman() then.

So, for me `queued_commands_P = pgcode;` here is absolutely required.